cmd/hyperkube/kubectl.go

```diff
@@ -19,21 +19,20 @@ package main
 import (
 	"os"
 
-	"k8s.io/kubernetes/cmd/kubectl/app"
+	"k8s.io/kubernetes/pkg/kubectl/cmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 )
 
 func NewKubectlServer() *Server {
+	cmd := cmd.NewKubectlCommand(cmdutil.NewFactory(nil), os.Stdin, os.Stdout, os.Stderr)
 	return &Server{
 		name:        "kubectl",
 		SimpleUsage: "Kubernetes command line client",
 		Long:        "Kubernetes command line client",
 		Run: func(s *Server, args []string) error {
-			os.Args = os.Args[1:]
-			if err := app.Run(); err != nil {
-				os.Exit(1)
-			}
-			os.Exit(0)
-			return nil
+			cmd.SetArgs(args)
```

Comment:
Depending on whether or not it's a softlink (`os.Args[0] == "hyperkube"`), it's either `os.Args[1:]` or `os.Args[2:]`. It's kind of nice to let the hyperkube server deal with that logic and sub-servers can just assume that their args list starts with the first relevant flag. This would easily support, for example, nesting servers in servers.